File: javascript/rna-transcription/rna-transcription.js
export const toRna = (dna) => {

  let rna = "";

  for (const c of dna) {

    switch (c) {
      case 'G':
        rna += 'C';
        break;
      case 'C':
        rna += 'G';
        break;
      case 'T':
        rna += 'A';
        break;
      case 'A':
        rna += 'U';
        break;
    }
  }

  return rna;
};

File: go/rna-transcription/rna_transcription.go
package strand

import "strings"

var mapping = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

// ToRNA convert a DNA strand into its RNA complement
func ToRNA(dna string) string {

	var sb strings.Builder

	for _, r := range dna {

		if val, ok := mapping[r]; ok {
			sb.WriteRune(val)
		}
	}

	return sb.String()
}
